lib/LaTeXML/Package/pgfmath.code.tex.ltxml

```diff
@@ -646,87 +651,133 @@ BEGIN {
   };
 
   $::RD_HINT = 1;
+  # $::RD_TRACE = 1;
+
+  # The PGF grammar is simple enough that can be parsed from left to right:
+  # once a part of the text has been parsed successfully, its meaning will not
+  # change. To ensure we do not waste energy on reparsing the same text, follow
+  #
+  # REQUIREMENT: alternations MUST be disambiguated by the first character
+  #
+  # In other words, the parser must never backtrack. This guarantees that
+  # successful matches and effectful computations such as 'rand' are not
+  # repeated multiple times.
 
   # Why can't I manage to import a few functions to be visible to the grammar actions?
   # NOTE Not yet done: quoted strings, extensible functions
   $PGFMATHGrammarSpec = << 'EoGrammar';
 #  {BEGIN { use LaTeXML::Package::Pool; }}
 #  { use LaTeXML::Package::Pool; }
 #  { LaTeXML::Package::Pool->import(qw(pgfmath_apply)); }
-  <skip:'[\s\{\}]*'>            # braces ignored during parse...
-  formula :
-    expr /\?/ expr /:/ expr { ($item[1] ? $item[3] : $item[5]); }
-  | expr CMP expr           { LaTeXML::Package::Pool::pgfmath_apply($item[2], $item[1], $item[3]); }
-  | expr
-
-expr :
-     term (ADDOP term { [$item[1],$item[2]]; })(s?)
-          { LaTeXML::Package::Pool::pgfmath_leftrecapply($item[1],map(@$_,@{$item[2]})); }
-
-  term :
-     factor (MULOP factor { [$item[1],$item[2]]; })(s?)
-          { LaTeXML::Package::Pool::pgfmath_leftrecapply($item[1],map(@$_,@{$item[2]})); }
-
-   # addPostfix[$base] ; adds any following sub/super scripts to $base.
-   addPostfix :
-          /^\Z/ { $arg[0];}   # short circuit!
-        | POSTFIX          addPostfix[LaTeXML::Package::Pool::pgfmath_apply($item[1],$arg[0])]
-        | { $arg[0]; }
-  factor : simplefactor /\^/ simplefactor { $item[1] ** $item[3]; }
-    | simplefactor addPostfix[$item[1]]
-
-  simplefactor :
-      /\(/ formula /(?:\)|^\Z)/     { $item[2]; } # Let unclosed () succeed at end?
-    | PREFIX simplefactor   { LaTeXML::Package::Pool::pgfmath_apply($item[1],$item[2]); }
-    | SIZER /\(/ QTEX /\)/ { LaTeXML::Package::Pool::pgfmath_sizer($item[1], $item[3]); }
-    | FUNCTION /\(/ formula (/,/ formula { $item[2]; })(s?) /\)/
-           { LaTeXML::Package::Pool::pgfmath_apply($item[1], $item[3], @{$item[4]}); }
-    | FUNCTION simplefactor
-           { LaTeXML::Package::Pool::pgfmath_apply($item[1], $item[2]); }
-    | FUNCTION0                   { LaTeXML::Package::Pool::pgfmath_apply($item[1]); }
-    | NUMBER UNIT          { LaTeXML::Package::Pool::pgfmath_convert($item[1],$item[2]); }
-    | NUMBER REGISTER      { LaTeXML::Package::Pool::pgfmath_apply('*', $item[1], $item[2]); }
+
+<skip:'[\s\{\}]*'>            # braces ignored during parse...
+
+formula : predicate conditional[$item{predicate}](?)
+  { $item[2][0] // $item{predicate}; }
```

Comment:
Perl 5.28-5.34 interprets this line *and the following ones until comparison* as part of the action, hence the failures.

Reply:
The problem seems to be caused by `//`, I replaced it with `defined ? :`.